### DISCUSSION TITLE
Achievements metrics shows (Unexpected Error)

### DISCUSSION THREAD

**Discussion by orsnaro** [lowlighter/metrics#1478] [Q&A] [answered]
![image](https://github.com/lowlighter/metrics/assets/77698890/83ac4ad0-97a0-4b77-9ea8-b2ae3f034352)

[metrics code in my `metrics_action.yml`](https://github.com/orsnaro/orsnaro/blob/main/.github/workflows/metrics_action.yml#L73C11-L73C11)

**Comment by Ankitz007** [comment]
Hi. Please replace the following in your action workflow file for achievements workflow -
` - uses: lowlighter/metrics@latest`

to this - 
`- uses: lowlighter/metrics@96dde98a034ab75fec7884f9161bc12759275d36`

FYI, it's the hash for the commit after which @lowlighter merged the workaround after recent changes in github search. (#1487)

**Reply by SupianIDz** [reply]
this solves the problem, thank you.

**Reply by lulunac27a** [reply]
I tried `- uses: lowlighter/metrics@latest`, `- uses: lowlighter/metrics@master` and `- uses: lowlighter/metrics@96dde98a034ab75fec7884f9161bc12759275d36` in main.yml, but the achievements are not working anymore due to unexpected error

**Reply by orsnaro** [reply]
yes , Achievements started showing the same error again on my GitHub overview also

**Reply by lowlighter** [reply, marked as answer]
This should be fixed with #1532 that landed a few hours ago
The workaround broke because it seems the regex didn't match anymore

But to make it really reliable, I decided to fetch the total values directly using GitHub GraphQL API so it should be almost failproof now since it's now using the "official way" to retrieve these data. I think it wasn't using the API before because it wasn't exposed at the time when the plugin was first written.

Currently builds and tests seems to pass, so hopefully if the rest of the CI is all green I'll make a new release tomorrow

**Reply by xero-lib** [reply]
It seems that this issue has returned, and is even affecting @lowlighter themselves' profile page.

**Reply by Ankitz007** [reply]
I guess this works for now: https://github.com/Ankitz007/Ankitz007/commit/566ba8e5aa7800ebc1bc27d7fa8d8fefa9683c5c

**Reply by xero-lib** [reply]
> I guess this works for now: [Ankitz007/Ankitz007@566ba8e](https://github.com/Ankitz007/Ankitz007/commit/566ba8e5aa7800ebc1bc27d7fa8d8fefa9683c5c)

You're a life-saver
